src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sortField = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  let preparedPeople = [...people];
+
+  if (sex) {
+    preparedPeople = preparedPeople.filter(person => person.sex === sex);
+  }
+
+  if (centuries.length > 0) {
+    const getCentury = (person: Person) => Math.ceil(person.born / 100);
+
+    preparedPeople = preparedPeople.filter(
+      person => centuries.includes(
+        getCentury(person).toString(),
+      ),
+    );
+  }
+
+  if (query) {
+    const lowerQuery = query.toLocaleLowerCase();
+
+    preparedPeople = preparedPeople.filter(person => {
+      return [
+        person.name,
+        person.motherName || '',
+        person.fatherName || '',
+      ]
+        .join('\n')
+        .toLocaleLowerCase()
```

Comment:
let's move this logic to some function, it will look better and be more readable